module/Checker/src/Service/Renewal.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Checker\Service;
+
+use Application\Service\Email as EmailService;
+use Checker\Mapper\Member as MemberMapper;
+use Database\Mapper\ActionLink as ActionLinkMapper;
+use Database\Model\ActionLink as ActionLinkModel;
+use DateInterval;
+use DateTime;
+use Laminas\View\Model\ViewModel;
+use Laminas\View\Renderer\PhpRenderer;
+
+/**
+ * Renewal class that takes care of renewing graduates
+ * and converting memberships to graduates
+ */
+class Renewal
+{
+    /**
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification
+     */
+    public function __construct(
+        private readonly ActionLinkMapper $actionLinkMapper,
+        private readonly MemberMapper $memberMapper,
+        private readonly EmailService $emailService,
+        private readonly PhpRenderer $renderer,
+        private readonly array $config,
+    ) {
+    }
+
+    /**
+     * Create an actionlink and send emails to expiring graduates
+     * Emails are sent 45 days before expiry
+     * A limit of 10 graduates is used; e.g. on a cronjob each hour this would mean 250 per day
+     * Limiting to make sure the secretary does not get overwhelmed with questions regarding renewal.
+     */
+    public function sendRenewalGraduates(): void
+    {
+        $expiresWithin = new DateTime();
+        $expiresWithin->add(new DateInterval('P45D'));
+        $limit = 10;
+        $graduates = $this->memberMapper->getExpiringGraduates($expiresWithin, $limit);
+
+        foreach ($graduates as $graduate) {
+            $actionLink = $this->actionLinkMapper->createByMember($graduate);
+            $this->sendRenewalEmail($actionLink);
+        }
+    }
+
+    private function sendRenewalEmail(ActionLinkModel $link): void
+    {
+        $body = $this->render(
+            'email/graduate-renewal',
+            [
+                'firstName' => $link->getMember()->getFirstName(),
+                'currentExpiration' => $link->getCurrentExpiration(),
+                'newExpiration' => $link->getNewExpiration(),
+                'url' => $this->config['application']['public_url'] . '/member/renew/' . $link->getToken(),
+                //TODO: If global config exists, we should make the secretary a global config option
+            ],
+        );
+
+        $this->emailService->sendEmailTemplate(
```

Comment:
This does not have to change but at some point it might be nice to make this named arguments.

Reply:
Still stands, but resolving to clear blockages.